src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1490,6 +1490,82 @@ Copyright (C) Microsoft Corporation. All rights reserved.
   </Target>
 
   <!--
+    ====================================================================================
+                                        _GetProjectReferenceTargetFrameworkProperties
+
+    Builds the GetTargetFrameworkProperties target of all existent project references,
+    passing $(TargetFrameworkMoniker) as $(ReferringTargetFramework) and sets the
+    SetTargetFramework metadata of the project reference to the value that is returned.
+
+    This allows a cross-targeting project to select how it should be configured to
+    build against the most appropriate target for the referring target framework.
+
+    ======================================================================================
+  -->
+  <Target Name="_GetProjectReferenceTargetFrameworkProperties" Outputs="%(_MSBuildProjectReferenceExistent.Identity)">
```

Comment:
Ok, I have an approach that allows building the refs in parallel. But I'm not sure it's actually better--it requires adding redundant metadata to `_MSBuildProjectReferenceExistent` (to allow for [shared-metadata batching](https://msdn.microsoft.com/en-us/library/ms171474.aspx#Anchor_1) on the path of the project):

```XML
  <Target Name="b">
    <ItemGroup>
      <_MSBuildProjectReferenceExistent>
        <MSBuildSourceProjectFile>%(FullPath)</MSBuildSourceProjectFile>
      </_MSBuildProjectReferenceExistent>
    </ItemGroup>

    <MSBuild
        Projects="@(_MSBuildProjectReferenceExistent)"
        Targets="GetTargetFrameworkProperties"
        BuildInParallel="$(BuildInParallel)">
      <Output TaskParameter="TargetOutputs" ItemName="_ProjectReferenceTargetFrameworkConfiguration" />
    </MSBuild>

    <ItemGroup>
      <_MSBuildProjectReferenceExistent Condition="'%(MSBuildSourceProjectFile)' != '' and '@(_MSBuildProjectReferenceExistent->'%(FullPath)')' == '@(_ProjectReferenceTargetFrameworkConfiguration->'%(MSBuildSourceProjectFile)')'">
        <Setting>@(_ProjectReferenceTargetFrameworkConfiguration->'%(Identity)')</Setting>
      </_MSBuildProjectReferenceExistent>

      <Test Include="@(_MSbuildprojectreferenceexistent)" />
    </ItemGroup>
  </Target>
```

Salient features:
* Add `MSBuildSourceProjectFile` metadata to `_MSBuildProjectReferenceExistent`
* Build in parallel, using an **Item** to catch the output (it'll automatically have `MSBuildSourceProjectFile`)
* Shared-metadata-batch on that to get refs one at a time
* set the desired metadata from the (also now one-at-a-time) output of the query

I'm not sure which use of obscure hard-to-reason-about batching features is better. This way is probably slightly more performant (because of parallelism in the remote builds). @cdmihai @AndyGerlicher @jeffkl--thoughts?